### PR TITLE
10795 Changed Profile Notifications Checkbox from environment flag to feature flag

### PR DESCRIPTION
**Pull request by elwoodva**
## DescriptionThe VA.gov site administrator wants to be able to control who sees the 
Notifications checkbox on the profile page so that it can be turned off or on or only shown to a percentage of Veterans.

## Testing done

Local testing.  Testing on staging is next.

## Acceptance criteria
- [ ] When the feature flag is enabled, all qualifying Veterans will be able to see the checkbox below the Mobile Phone number on the profile page and on the edit modal for the mobile phone number on the profile page.
- [ ] When the feature flag is disabled, no Veterans will be able to see the checkbox below the Mobile Phone Number anywhere.

### CODE REVIEW

**Review by talbotmj — APPROVED** [review]

File: src/platform/utilities/feature-toggles/featureFlagNames.js
Comment (erikphansen) [inline src/platform/utilities/feature-toggles/featureFlagNames.js]: could you make this flag name a little more specific? "notifications" is pretty general. maybe something like "profileShowTextMessageNotifications" or "profileShowNotificationsViaTextMessage"?

File: src/platform/user/profile/vet360/components/base/TransactionPending.jsx
Comment (erikphansen) [inline src/platform/user/profile/vet360/components/base/TransactionPending.jsx]: Could you change this from `'../../selectors'` to `'vet360/selectors'`? This is optional but I find it a little bit clearer and easier to understand vs. trying to figure out which directory is two directories up.

File: src/platform/user/profile/vet360/components/MobilePhone.jsx
Comment (erikphansen) [inline src/platform/user/profile/vet360/components/MobilePhone.jsx]: this doesn't need to be converted to a React.Component. I think you could just remove the `export default` and pass the props arg to the `MobilePhone` function...

File: src/platform/user/profile/vet360/components/MobilePhone.jsx
Comment (erikphansen) [inline src/platform/user/profile/vet360/components/MobilePhone.jsx]: ...and then remove the `this` from in front of `props`. I don't feel super strongly about this, but it seems better to keep it a functional component since it's not using any lifecycle methods.

File: src/platform/user/profile/vet360/components/MobilePhone.jsx
Comment (elwoodva) [reply src/platform/user/profile/vet360/components/MobilePhone.jsx]: I want to leave it this way because I don't want to deal with passing this flag in as a prop JUST to mobile phone.

File: src/platform/user/profile/vet360/components/MobilePhone.jsx
Comment (elwoodva) [reply src/platform/user/profile/vet360/components/MobilePhone.jsx]: Thanks!

**Review by erikphansen — APPROVED** [review]

File: src/platform/user/profile/vet360/components/MobilePhone.jsx
Comment (erikphansen) [inline src/platform/user/profile/vet360/components/MobilePhone.jsx]: You've got some failing unit tests. Without digging into them too deeply, you'll probably need to make a few adjustments.

1. Export the non-connected component from this file. So `export { MobilePhone };` should do it.
2. Then in any test file that is using this component, make sure it's using the non-connected component (the new named export) rather than the connected component that's now the default export from this file.

Comment (elwoodva) [reply src/platform/user/profile/vet360/components/MobilePhone.jsx]: I did the export.  I don't think any particular tests are using this (that I can find).  (But I'm having difficulty accessing jenkins to see what's going on in particular.)

Comment (erikphansen) [reply src/platform/user/profile/vet360/components/MobilePhone.jsx]: I'm guessing if you run the tests locally you'll get the same failures that Jenkins is getting.

File: src/platform/user/profile/vet360/containers/TimedAlertBox.jsx
Comment (erikphansen) [inline src/platform/user/profile/vet360/containers/TimedAlertBox.jsx]: maybe move this up into the check on line 10? so `if (this.props.startTimer && !this.intervalId)`? Although checking for `this.intervalId` seems unnecessary in this case because it will always be undefined when the component mounts?

File: src/platform/user/profile/vet360/containers/TimedAlertBox.jsx
Comment (erikphansen) [inline src/platform/user/profile/vet360/containers/TimedAlertBox.jsx]: should a default value for the `interval` prop be set?

Comment (erikphansen) [reply src/platform/user/profile/vet360/containers/TimedAlertBox.jsx]: also, just for the sake of semantics, could you use `setTimeout` instead since this is only going to call the callback once

File: src/platform/user/profile/vet360/containers/TimedAlertBox.jsx
Comment (erikphansen) [inline src/platform/user/profile/vet360/containers/TimedAlertBox.jsx]: If you switch to `setTimeout` you can remove this.

**Review by erikphansen — DISMISSED** [review]

File: src/platform/user/profile/vet360/containers/TimedAlertBox.jsx
Comment (elwoodva) [reply src/platform/user/profile/vet360/containers/TimedAlertBox.jsx]: It's not.  It was never supposed to be committed.  I have removed it.

**Comment by elwoodva** [comment]
@rianfowler By peer review does that mean someone from my own team?  Michael Talbot is from my team.

**Comment by elwoodva** [comment]
@erikphansen Does my explanation satisfy you?  I have removed the file that was accidentally included.  Would you be able to approve this pull request now?

File: src/platform/user/profile/vet360/tests/components/Transaction.unit.spec.jsx
Comment (erikphansen) [inline src/platform/user/profile/vet360/tests/components/Transaction.unit.spec.jsx]: adding this class so we can use the `'div.content'` selector down below...

File: src/platform/user/profile/vet360/tests/components/Transaction.unit.spec.jsx
Comment (erikphansen) [inline src/platform/user/profile/vet360/tests/components/Transaction.unit.spec.jsx]: this is slightly redundant since we are testing that the call to `html()` above renders `'Children'` but I wanted to add this test here as proof that it works so when we use the same `component.find('div.content')` down below we know that one is working as well.

File: src/platform/user/profile/vet360/tests/components/Transaction.unit.spec.jsx
Comment (erikphansen) [inline src/platform/user/profile/vet360/tests/components/Transaction.unit.spec.jsx]: this is a brand new `expect` that improves the test coverage a bit

File: src/platform/user/profile/vet360/tests/components/Transaction.unit.spec.jsx
Comment (erikphansen) [inline src/platform/user/profile/vet360/tests/components/Transaction.unit.spec.jsx]: we can't call `html()` anymore since the Transaction component will try to render out the Redux-connected TransactionPending component. But we can use `find()` instead.

**Review by erikphansen — CHANGES_REQUESTED** [review]
looks good! just remove that duplicate selector and i think we're good to go. I also added comments on the tests I adjusted.

**Comment by erikphansen** [comment]
@elwoodva Rian saw that I was in the middle of reviewing this PR after Michael had approved it, which is why he added the Peer Review Needed label (since there's no way to switch a PR back to being a Draft).

**Review by rianfowler — APPROVED** [review]
I added the label because the PR needs an approval from @erikphansen. I think when you request changes, you effectively block the merge until it has your approval. We usually don't approve it until it's ready to be merged. 

Screen Shot 2019-10-29 at 11 33 04 AM

**Comment by elwoodva** [comment]
Thanks @rianfowler

File: src/applications/personalization/profile360/selectors.js
Comment (elwoodva) [reply src/applications/personalization/profile360/selectors.js]: @erikphansen I have removed the feature flag from the platform selectors and have updated the code to import the one from the application selectors.

**Review by erikphansen — APPROVED** [review]

**Pull request merged**
